h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderBuilder.java

```diff
@@ -29,38 +29,43 @@ public TargetEncoderBuilder(final boolean startupOnce) {
     super(new TargetEncoderModel.TargetEncoderParameters(), startupOnce);
   }
 
-  @Override
-  public void init(boolean expensive) {
-    super.init(expensive);
-  }
-
   private class TargetEncoderDriver extends Driver {
     @Override
     public void computeImpl() {
-      // In case ModelBuilder is registered with default model parameters ( i.e. without training frame etc.) we want to init it at this stage as well
-      if(_startUpOnceModelBuilder) init(false);
-
-      final int numColsRemoved = hasFoldCol() ? 2 : 1; // Response is always at the last index, fold column is on the index before.
-
-      final String[] encodedColumns = Arrays.copyOf(train().names(), train().names().length - numColsRemoved);
-      TargetEncoder tec = new TargetEncoder(encodedColumns);
-
-      Scope.untrack(train().keys());
-
-      IcedHashMapGeneric<String, Frame> _targetEncodingMap = tec.prepareEncodingMap(train(), _parms._response_column, _parms._fold_column);
-
-      // Mean could be computed from any encoding map as response column is shared
-      double priorMean = tec.calculatePriorMean(_targetEncodingMap.entrySet().iterator().next().getValue());
-
-      for(Map.Entry<String, Frame> entry: _targetEncodingMap.entrySet()) {
-        Frame frameWithEncodingMap = entry.getValue();
-        Scope.untrack(frameWithEncodingMap.keys());
+      _targetEncoderModel = null;
+      try {
+        init(true);
+
+        final int numColsRemoved = hasFoldCol() ? 2 : 1; // Response is always at the last index, fold column is on the index before.
+        final String[] encodedColumns = Arrays.copyOf(train().names(), train().names().length - numColsRemoved);
+        TargetEncoder tec = new TargetEncoder(encodedColumns);
+
+        TargetEncoderModel.TargetEncoderOutput emptyOutput =
+                new TargetEncoderModel.TargetEncoderOutput(TargetEncoderBuilder.this, new IcedHashMapGeneric<>(), Double.NaN);
+        TargetEncoderModel model = new TargetEncoderModel(dest(), _parms, emptyOutput, tec);
+        _targetEncoderModel = model.delete_and_lock(_job); // and clear & write-lock it (smashing any prior)
```

Comment:
@michalkurka thanks! Now I see that model was not locked. I was confused by lock on a grid here https://github.com/h2oai/h2o-3/blob/97624413c5bd154076716fc15622d8193933c9b8/h2o-core/src/main/java/hex/grid/GridSearch.java#L114 . I thought that it is locking models as well but now I see how naive this  thought was :)